File: cpp/563/Binary_Tree_Tilt.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int findTilt(TreeNode* root) {
        int tilt = 0;
        findTilt(root, tilt);
        return tilt;
    }
    
    int findTilt(TreeNode *root, int &tilt){
        if(!root) return 0;
        int l = findTilt(root->left, tilt);
        int r = findTilt(root->right, tilt);
        tilt += abs(l-r);
        return l+r+root->val;
    }
};

File: python/563/Binary_Tree_Tilt.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def findTilt(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        res = [0]
        def dfs(root):
            if root:
                rl = dfs(root.left)
                rr = dfs(root.right)
                res[0] += abs(rl-rr)
                return root.val + rl + rr
            return 0
        dfs(root)
        return res[0]
        
